yara-sys/build.rs

```diff
@@ -1,48 +1,138 @@
+            .file(basedir.join("libyara/compiler.c"))
+            .file(basedir.join("libyara/endian.c"))
+            .file(basedir.join("libyara/exec.c"))
+            .file(basedir.join("libyara/exefiles.c"))
+            .file(basedir.join("libyara/filemap.c"))
+            .file(basedir.join("libyara/grammar.c"))
+            .file(basedir.join("libyara/hash.c"))
+            .file(basedir.join("libyara/hex_grammar.c"))
+            .file(basedir.join("libyara/hex_lexer.c"))
+            .file(basedir.join("libyara/lexer.c"))
+            .file(basedir.join("libyara/libyara.c"))
+            .file(basedir.join("libyara/mem.c"))
+            .file(basedir.join("libyara/object.c"))
+            .file(basedir.join("libyara/parser.c"))
+            .file(basedir.join("libyara/proc.c"))
+            .file(basedir.join("libyara/re.c"))
+            .file(basedir.join("libyara/re_grammar.c"))
+            .file(basedir.join("libyara/re_lexer.c"))
+            .file(basedir.join("libyara/rules.c"))
+            .file(basedir.join("libyara/scan.c"))
+            .file(basedir.join("libyara/scanner.c"))
+            .file(basedir.join("libyara/sizedstr.c"))
+            .file(basedir.join("libyara/stack.c"))
+            .file(basedir.join("libyara/stopwatch.c"))
+            .file(basedir.join("libyara/stream.c"))
+            .file(basedir.join("libyara/strutils.c"))
+            .file(basedir.join("libyara/threading.c"))
+
+            .file(basedir.join("libyara/modules.c"))
+            .file(basedir.join("libyara/modules/elf.c"))
+            .file(basedir.join("libyara/modules/math.c"))
+            .file(basedir.join("libyara/modules/pe.c"))
+            .file(basedir.join("libyara/modules/pe_utils.c"))
+            .file(basedir.join("libyara/modules/tests.c"))
+            .file(basedir.join("libyara/modules/time.c"))
+
+            .define("DEX_MODULE", "")
+            .file(basedir.join("libyara/modules/dex.c"))
+
+            .define("DOTNET_MODULE", "")
+            .file(basedir.join("libyara/modules/dotnet.c"))
+
+            .define("MACHO_MODULE", "")
+            .file(basedir.join("libyara/modules/macho.c"));
+
+        // Use correct proc functions
+        match std::env::var("CARGO_CFG_TARGET_OS").ok().unwrap().as_str() {
+            "windows"
+                => cc.file(basedir.join("libyara/proc/windows.c"))
+                    .define("USE_WINDOWS_PROC", ""),
+            "linux"
+                => cc.file(basedir.join("libyara/proc/linux.c"))
+                    .define("USE_LINUX_PROC", ""),
+            "macos"
+                => cc.file(basedir.join("libyara/proc/mach.c"))
+                    .define("USE_MACH_PROC", ""),
+            _
+                => cc.file(basedir.join("libyara/proc/none.c"))
+                     .define("USE_NO_PROC", ""),
+        };
+
+        match std::env::var("CARGO_CFG_TARGET_FAMILY").ok().unwrap().as_str() {
+            "windows"
+                => cc.define("NDEBUG", "1"),
```

Comment:
I am not familiar with the Yara build, but this variable should probably also be set on Linux, no? Or is it automatic?